### PR TITLE
APIGOV-22033 Handlers & marketplace updates

### PR DESCRIPTION
**Pull request by tsjohns9**

### CODE REVIEW

File: pkg/agent/handler/accessrequest.go [hunk 1]
```diff
@@ -47,18 +48,19 @@ func (h *accessRequestHandler) Handle(action proto.Event_Type, meta *proto.Event
 		return nil
 	}
 
+	log.Infof("%s event for AccessRequest", action.String())
```
Comment (jcollins-axway) [inline pkg/agent/handler/accessrequest.go:51]: This would be better as Debug

File: pkg/agent/handler/credential.go [hunk 2]
```diff
@@ -52,18 +46,19 @@ func (h *credentials) Handle(action proto.Event_Type, meta *proto.EventMeta, res
 		return nil
 	}
 
+	log.Infof("%s event for Credential", action.String())
```
Comment (jcollins-axway) [inline pkg/agent/handler/credential.go:49]: same

File: pkg/agent/handler/credential.go [hunk 3]
```diff
@@ -90,11 +85,17 @@ func (h *credentials) Handle(action proto.Event_Type, meta *proto.EventMeta, res
 	if cr.Status.Level == statusPending {
 		status, credentialData = h.prov.CredentialProvision(creds)
 		sec := app.Spec.Security
-		enc, err := newEncryptor(sec.EncryptionKey, sec.EncryptionAlgorithm, sec.EncryptionHash)
+		enc, err := util.NewEncryptor(sec.EncryptionKey, sec.EncryptionAlgorithm, sec.EncryptionHash)
```
Comment (jcollins-axway) [inline pkg/agent/handler/credential.go:88]: love this update

File: pkg/agent/handler/managedapplication.go [hunk 4]
```diff
@@ -38,18 +39,19 @@ func (h *managedApplication) Handle(action proto.Event_Type, meta *proto.EventMe
 		return nil
 	}
 
+	log.Infof("%s event for ManagedApplication", action.String())
```
Comment (jcollins-axway) [inline pkg/agent/handler/managedapplication.go:42]: same

File: pkg/apic/client.go [hunk 5]
```diff
@@ -88,8 +88,8 @@ type Client interface {
 	UpdateAccessControlList(acl *mv1a.AccessControlList) (*mv1a.AccessControlList, error)
 	CreateAccessControlList(acl *mv1a.AccessControlList) (*mv1a.AccessControlList, error)
 	UpdateAPIV1ResourceInstance(url string, ri *apiv1.ResourceInstance) (*apiv1.ResourceInstance, error)
-	CreateSubResourceScoped(scopeKindPlural, scopeName, resKindPlural, name, group, version string, subs map[string]interface{}) error
-	CreateSubResourceUnscoped(kindPlural, name, group, version string, subs map[string]interface{}) error
+	CreateSubResourceScoped(rm v1.ResourceMeta, subs map[string]interface{}) error
+	CreateSubResourceUnscoped(rm v1.ResourceMeta, subs map[string]interface{}) error
```
Comment (jcollins-axway) [inline pkg/apic/client.go:92]: Just making sure that the prev defs of these are not used anywhere outside the SDK

File: pkg/apic/provisioning/applicationrequest.go [hunk 6]
```diff
@@ -2,10 +2,10 @@ package provisioning
 
 // ApplicationRequest - interface for agents to use to get necessary application request details
 type ApplicationRequest interface {
+	// GetAgentDetailsValue return the value based on the key
+	GetAgentDetailsValue(key string) string
```
Comment (jcollins-axway) [inline pkg/apic/provisioning/applicationrequest.go:6]: To keep consistent with the others we should change this to GetApplicaitonDetailsValue

**Review by jcollins-axway — APPROVED** [review]

File: pkg/agent/handler/accessrequest.go (hunk 1 above)
Comment (tsjohns9) [reply pkg/agent/handler/accessrequest.go:51]: Yeah. They will be removed before we are done.

File: pkg/apic/client.go (hunk 5 above)
Comment (tsjohns9) [reply pkg/apic/client.go:92]: I didn't use it in any agents, so I don't think so.

File: pkg/apic/provisioning/applicationrequest.go (hunk 6 above)
Comment (tsjohns9) [reply pkg/apic/provisioning/applicationrequest.go:6]: Yes

**Review by jcollins-axway — APPROVED** [review]

**Pull request merged**